ocs_ci/ocs/ui/block_pool.py

```diff
@@ -173,3 +176,130 @@ def reach_pool_limit(self, replica, compression):
                     raise PoolStateIsUnknow(
                         f"pool {pool_name} is in unexpected state {pool_state}"
                     )
+
+    def check_ui_pool_efficiency_parameters_against_prometheus(
+        self, pool_name, expected_compression_saving
+    ):
+        """
+        Get pool compression efficiency from UI and compare them against prometheus metrics
+
+        Args:
+            pool_name (str): The pool name to be checked.
+            expected_compression_saving (str): The expected compression saving in string like "1.5 GiB"
+
+        Returns:
+            (bool): True if UI and prometheus parameters are equal to expected_compression_saving - else False.
+
+        """
+
+        # Navigate to pool page
+        self.navigate_installed_operators_page()
+        self.navigate_block_pool_page()
+        self.do_click([f"{pool_name}", By.LINK_TEXT])
+        self.page_has_loaded(retries=15)
+
+        # Get efficiency card values
+        eff_elements = self.driver.find_elements_by_class_name(
+            "ceph-storage-efficiency-card__item-text"
+        )
+        eff_value_array = []
+        for element in eff_elements:
+            eff_value_array.append(element.text)
+        compression_eligibility, compression_ratio, compression_saving = eff_value_array
+        compression_saving_value, compression_saving_unit = compression_saving.split(
+            " "
+        )
+        logger.info(
+            f"compression_saving_value {compression_saving_value} "
+            f"compression_saving_unit {compression_saving_unit}"
+        )
+        logger.info(
+            f"compression_eligibility {compression_eligibility} "
+            f"compression_ratio {compression_ratio} "
+            f" compression_saving {compression_saving}"
+        )
+
+        # Get pool id and query Prometheus
+        pool_id = get_pool_id(pool_name)
+        if pool_id is None:
+            raise PoolIdNotFound(f"Pool {pool_name} didn't return pool id")
+        new_prom = PrometheusAPI()
+        ceph_pool_compress_bytes_used_data = new_prom.query(
+            f'ceph_pool_compress_bytes_used{{pool_id="{pool_id}"}}'
```

Comment:
please save the query in ocs_ci/ocs/constants.py:1227

Reply:
I saw the queries in constant. None of them have variables inside them and mine have pool id.
Whomever will use them will have to know that he needs to format the string to insert the variable. Are you ok with that?

Reply:
yes, it is ok